### PR TITLE
Loading Indicators to Borrow/Read/Waitlist Buttons

### LINKED ISSUE #996: Add loading indicators to actions like borrow/waitlist
Some actions take a long time. We may (at least) want to change our mouse cursor to a loading time-glass to indicate loading.

### PR DESCRIPTION
**Pull request by nimbus98**
**Description**: What does this PR achieve? [refactor]
Added loading indicators using JS functions for read/borrow/waitlist buttons
Closes #996 

 **Technical**: What should be noted about the implementation?

__Borrow/Read butons that were present in the macros were only modified for this feature.__
For those buttons that have been loaded dynamically in JS scripts, I tried implementing it there too, but it doesn't seem to work for this implementation, (I'm talking about the buttons shown below)

I tried doing this implementation on `availability.js` file, where the buttons are loaded dynamically as a result of search, (by addding onclick attribute to call the respective functions i made from ol.js), but this doesn't seem to work when I test on these buttons (I'm guessing its because when I inspected the element from browser it had two onclick listeners and only one was being called, the one that was being called is this function
 ``` 
function(event) {
  var category_action = $(this).attr('data-ol-link-track').split('|');
  window.archive_analytics.ol_send_event_ping({
    'category': category_action[0],
    'action': category_action[1]
  });
}
```  
while it ignored the calling load function i made on ol.js )

**Testing**: Steps for reviewer to reproduce / verify this PR fixes the problem?

Click on the buttons shown below.

 **Evidence**: If this PR touches UI, please post evidence (screenshot) of it behaving correctly:

__Before:__

__After:__

__Before:__

__After:__

### CODE REVIEW

**Comment by mekarpeles** [comment]
Thank you for these contributions @nimbus98!! Will pull on dev.openlibrary.org for us to try out

**Comment by jdlrobson** [comment]
I personally find it strange to see the spinner overlapping some text and appearing clickable. It's a little bit confusing as it's not clear what's loading.

Consider making the button translucent and hiding the label while loading it. Alternatively we could drop the spinner and cover the entire list with a translucent "shield" that says "checking book availability ". I imagine if there are more than one button loading in this fashion the page will look very noisy. Can you share a screenshot showing more than one being looked up?

**Comment by nimbus98** [comment]
> These would be better in ol.js like they were both. Otherwise we will have to maintain them in both these files... let's have one single point of truth!

I did that first! but jslint gave me errors during testing, saying that the load functions were called but not used. Theres a fix for that by just commenting this below in ol.js
```
// Code here will be linted with JSHint.
/* jshint ignore:start */
// Code here will be ignored by JSHint.
/* jshint ignore:end */
```
That should prevent the errors, but I didn't do it thinking it would be a bad programming practice.

**Comment by nimbus98** [comment]
> I personally find it strange to see the spinner overlapping some text and appearing clickable. It's a little bit confusing as it's not clear what's loading.
> 
> Consider making the button translucent and hiding the label while loading it. Alternatively we could drop the spinner and cover the entire list with a translucent "shield" that says "checking book availability ". I imagine if there are more than one button loading in this fashion the page will look very noisy. Can you share a screenshot showing more than one being looked up?

So would you like to change it to a text based thing then? I can test that out as well.
Yes sure, I'll send a screenshot asap today :)

**Comment by jdlrobson** [comment]
> I did that first! but jslint gave me errors during testing, saying that the load functions were called but not used

We do this already making clear to document where the function is used. See
https://github.com/internetarchive/openlibrary/blob/master/openlibrary/plugins/openlibrary/js/ol.js#L74 for an example.

We disable the explicit rule.. not everything!

**Comment by nimbus98** [comment]
> > I did that first! but jslint gave me errors during testing, saying that the load functions were called but not used
> 
> We do this already making clear to document where the function is used. See
> https://github.com/internetarchive/openlibrary/blob/master/openlibrary/plugins/openlibrary/js/ol.js#L74 for an example.
> 
> We disable the explicit rule.. not everything!

Wow, did not know about this!
Thank you ^^, I'll push an update to the code asap

**Comment by nimbus98** [comment]
@jdlrobson 
I made the changes, but there seems to be a problem with the size of CSS build file? Can someone help fix that?

**Comment by nimbus98** [comment]
@jdlrobson  your suggested changes are done and they work fine
one problem during testing, is that I get this error:
 FAIL  static/build/page-plain.css: 18.81KB > maxSize 18.8KB (gzip) 

I pulled cham

**Comment by jdlrobson** [comment]
> AIL static/build/page-plain.css: 18.81KB > maxSize 18.8KB (gzip)
That is telling you your changes have INCREASED the amount of CSS loaded on the critical path. This can lead to a performance degradation so should be avoided wherever possible.
Instead, let's move your new styles to a file buttonCta--js.less and load it inside `static/css/js-all.less`. This means that CSS will only get loaded via JavaScript.

**Comment by nimbus98** [comment]
It is done :)

File: static/css/components/buttonCta--js.less [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+     &--load {
+       background-color: @primary-blue !important;
+       background: url(../../../static/images/indicator.gif) center center no-repeat;
+       opacity: 0.6;
+     }
+   }
+   &--unavailable {
+     background-color: @orange;
+     &--load {
+       background-color: @orange !important;
```
Comment (jdlrobson) [inline static/css/components/buttonCta--js.less]: I think only the --load rules should be needed in this file!

Comment (nimbus98) [reply static/css/components/buttonCta--js.less]: My bad!
Didn't realize that sooner.
Changes have been made.

**Comment by nimbus98** [comment]
Changed and tested, sorry about that, hope only the necessary is present!

**Comment by jdlrobson** [comment]
Looks great! thank you!
I'm sorry this code is still evidently so clumsy to work with. I'd be super grateful for any input you can provide on fixing this.

I've started a project on https://github.com/internetarchive/openlibrary/projects/10?add_cards_query=is%3Aopen for modernizing the JS which will hopefully make clearer how and where JS is loading!

**Comment by nimbus98** [comment]
@jdlrobson 
Thank you! I'm more than happy to help out with documenting this on the wiki as mentioned on Slack :)
and yes, I've taken a look there as well! ^^

**Comment by mekarpeles** [comment]
Thank you for sticking with it!! Congrats on this PR

**Pull request merged**